Packs/NutanixHypervisor/Integrations/NutanixHypervisor/NutanixHypervisor.py

```diff
@@ -0,0 +1,1017 @@
+            raise DemistoException('Unauthorized - make sure you have the right credentials')
+        if e.message and 'Error in API call [404] - NOT FOUND' in e.message:
+            raise DemistoException('''Page not found - make sure 'Server URL' parameter is correct''')
+        raise e
+
+
+def fetch_incidents_command(client: Client, params: Dict):
+    """
+    Wrapper function that calls client fetch_incidents function with last run and demisto params.
+    Updates the new run, and uploads incidents to Demisto.
+    Args:
+        client (Client): The client to perform Nutanix API request and fetch incidents.
+        params (Dict): Demisto params.
+
+    Returns:
+        Fetches incidents to Demisto.
+    """
+    last_run = demisto.getLastRun()
+    incidents, next_run = client.fetch_incidents(params, last_run)
+    demisto.setLastRun(next_run)
+    demisto.incidents(incidents)
+
+
+def nutanix_hypervisor_hosts_list_command(client: Client, args: Dict):
+    """
+    Gets a list all physical hosts configured in the cluster by Nutanix service.
+    Possible filters:
+    - page: The offset page to start retrieving hosts.
+            When page is specified, limit argument is required, else Nutanix service returns an error
+            "Page number cannot be specified without count".
+    - limit: The number of hosts to retrieve, has to be positive.
+    - filter: Retrieve hosts that matches the filters given.
+              - Each filter is written in the following way: filter_name==filter_value or filter_name!=filter_value.
+              - Possible combinations of OR (using comma ',') and AND (using semicolon ';'), for Example:
+                storage.capacity_bytes==2;host_nic_ids!=35,host_gpus==x is parsed by Nutanix the following way:
+                Return all hosts s.t (storage.capacity_bytes == 2 AND host_nic_ids != 35) OR host_gpus == x.
+
+    In case response was successful, response will be a of list of hosts details.
+
+    Args:
+        client (Client): Client object to perform request.
+        args (Dict): Demisto arguments.
+
+    Returns:
+        CommandResults.
+    """
+    filter_ = args.get('filter')
+    limit = arg_to_number(args.get('limit', 50))
+    page = arg_to_number(args.get('page'))
+    verbose = get_optional_boolean_arg(args, 'verbose')
+
+    raw_response = client.get_nutanix_hypervisor_hosts_list(filter_, limit, page)
+    raw_outputs = raw_response.get('entities')
+
+    if verbose:
+        outputs = [{k: v for k, v in raw_output.items()
+                    if k not in NUTANIX_HOST_FIELDS_TO_REMOVE}
+                   for raw_output in raw_outputs]
+    else:
+        outputs = [{k: v for k, v in raw_output.items()
+                    if k in HOST_FIELDS_NOT_VERBOSE}
+                   for raw_output in raw_outputs]
+
+    add_iso_entries_to_dict(outputs)
+
+    final_outputs = [remove_empty_elements(output) for output in outputs]
```

Comment:
Let's add a function that does all these sanitization WDYT? it will receive a list to ignore of key such as HOST_FIELDS_NOT_VERBOSE or an empty one and will do al of this.

Reply:
@ShahafBenYakir can do that, but it gets a little messy in a function because you need many args
- verbose arg
- raw_outputs
- fields to remove (if verbose)
- fields to include (if not verbose)

i think it will be more confusing for someone to read than keep it inside the command function

Reply:
First of all, only now I see you have one list which is things you want to ignore and the other is things you want to have. for simplicity, I think we should be consistent and use one type (the smaller one).

Then there will be no need for all
- verbose arg
- fields to remove (if verbose)
- fields to include (if not verbose)


And you could have only:
- raw_outputs
- fields to remove (This will default to `[]`, for cases you don't want to remove anything)

So for example:
```
keys_to_ignore = some_list
if verbose:
    keys_to_ignore = bigger_list

new_function_to_sanitize_output(outputs, keys_to_ignore)
```

And if you don't want to remove any key:
```
new_function_to_sanitize_output(outputs)
```

Reply:
Getting the keys_to_ignore is a bit messy if verbose = False.
Because i will need to iterate through all keys in the outputs and do something like
if host_list_is_not_empty:
   keys_to_ignore = {k for k in hosts_list[0].keys() if k not in HOST_FIELDS_NOT_VERBOSE}
   
What do you think? better this way?
@ShahafBenYakir

Reply:
You do it anyway when you do this no?
```
outputs = [{k: v for k, v in raw_output.items()
                    if k in HOST_FIELDS_NOT_VERBOSE}
                   for raw_output in raw_outputs]
```

Reply:
it goes "if k in HOST_FIELDS_NOT_VERBOSE"  
I have the list of the fields to be sent hard coded (list of around 28-30 fields)
If i want to have the list of "fields not exclude when not verbose", ill have to hard code a list of 120+ fields, or do something like keys_to_ignore = {k for k in hosts_list[0].keys() if k not in HOST_FIELDS_NOT_VERBOSE}